pkg/kubelet/kubelet_pods.go

```diff
@@ -435,6 +435,21 @@ func (kl *Kubelet) GetPodCgroupParent(pod *v1.Pod) string {
 	return cgroupParent
 }
 
+// GetHostNameForKernel gets hostname value to set in the hostname field (the nodename field of struct utsname) of the pod.
+func (kl *Kubelet) GetHostNameForKernel(hostname string, hostDomainName string, hostnameFQDN bool) (string, error) {
+	kernelhostname := hostname
+	if len(hostDomainName) > 0 && hostnameFQDN {
+		fqdnHostname := fmt.Sprintf("%s.%s", hostname, hostDomainName)
```

Comment:
s/fqdnHostname/fqdn/g

looks we are form FQDN using pod's hostname and domain

Reply:
fixing